core/commands/config.go

```diff
@@ -524,33 +518,57 @@ func replaceConfig(r repo.Repo, file io.Reader) error {
 		return errors.New("private key in config was not a string")
 	}
 
-	cfg.Identity.PrivKey = pkstr
+	newCfg.Identity.PrivKey = pkstr
+
+	// Handle Pinning.RemoteServices (API.Key of each service is a secret)
 
-	// Handle Pinning.RemoteServices (ApiKey of each service is secret)
-	// Note: these settings are opt-in and may be missing
+	newServices := newCfg.Pinning.RemoteServices
+	oldServices, err := getRemotePinningServices(r)
+	if err != nil {
+		return tryRemoteServiceApiErr
```

Comment:
why this error? It seems like an error relating to the remote pinning service config being broken.

Reply:
Yes, this should be `return err` or a wrapped one.